### ISSUE TITLE
Any result on larger-scale dataset?

### ISSUE THREAD

**Issue by ymchen7** [epfml/powersgd#1]
Hi, nice work on gradient compression. I have tried Rank 4 reducer on imagenet, and the top1 accuracy drops about 3.5% (76%->72.48%), the hyper-paremeters are almost the same with imagenet example except the lr is proportional increased with batchsize(8K). Is it any workaround here? From my trial, it seems the generalization degrades a lot on a large-scale large-batch-size setting(while it is the exact case that the communication becomes bottleneck). Looking forward to hear your suggestions.

BTW, in the 'rankkreducer', why only Q matrix divided by the world-size, is it a typo or unnecessary since we orthogonalizes P matrix?

**Comment by tvogels** [comment]
Hello! Thanks for trying out the code. You probably need a higher rank for ImageNet to be train in the same number of epochs. For the large transformer model in the appendix of our paper, we went up to rank 16 to get reasonable results. I would be curious what the outcomes of your experiments are for higher ranks. 

You are right about the division by world-size. Because the p-matrices are [orthonormalized anyway](https://github.com/epfml/powersgd/blob/master/gradient_reducers.py#L764-L765), scaling by a scalar wouldn't change the output. Minor optimization ...

**Comment by ymchen7** [comment]
> Hello! Thanks for trying out the code. You probably need a higher rank for ImageNet to be train in the same number of epochs. For the large transformer model in the appendix of our paper, we went up to rank 16 to get reasonable results. I would be curious what the outcomes of your experiments are for higher ranks.
> 
> Note that the communication/computation is quite a bit lower for ImageNet than for Cifar, due tot he large image size. A lower ratio would imply you probably benefit less from the compression. We didn't continue with ImageNet experiments after we saw that communication wasn't really a bottleneck there on our setup.
> 
> You are right about the division by world-size. Because the p-matrices are [orthonormalized anyway](https://github.com/epfml/powersgd/blob/master/gradient_reducers.py#L764-L765), scaling by a scalar wouldn't change the output. Minor optimization ...

Thanks for response. I'll try a larger rank and update the result here.

**Comment by ymchen7** [comment]
rank 8/16 top 1 accuracy 74.58/75.33. I think with proper hyperparameter tuning and tricks for large batch-size training, it could maintain the same acc on rank 16 somehow.

**Comment by tvogels** [comment]
@ymchen7 great, thanks very much for running these experiments. 

Tuning the hyper-parameters might do the trick, but if you want to avoid this, I expect that an even higher rank would achieve the same (with the normal hyperparameters). I guess the choice depends on the specific setting and timings.

Is the target 76% you mentioned achieved by a single worker or with the same number of workers you are using for these experiments?

**Comment by ymchen7** [comment]
@tvogels All the hyper-parameters are the same, batchsize 8k with 32 workers. I am wondering a layer wise rank (like LARS) may lead to a performance boost.

**Comment by tvogels** [comment]
Hmm .. yeah, that sounds like it could help. 

I'll close this issue now, but feel free to keep commenting.
